From 0db9bbd8d7ec1242e5c7c7876919e61402532075 Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 29 Apr 2020 00:15:49 +0200
Subject: [PATCH 18/25] Revert "Ignore broken camcorder profiles"

This reverts commit 5caabfcd03ca00211a0419ec3c04cff1a4efcf9e.

Will use upstream's better alternative
---
 media/libmedia/MediaProfiles.cpp | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/media/libmedia/MediaProfiles.cpp b/media/libmedia/MediaProfiles.cpp
index 0944c973d..08d0199a0 100644
--- a/media/libmedia/MediaProfiles.cpp
+++ b/media/libmedia/MediaProfiles.cpp
@@ -341,12 +341,10 @@ MediaProfiles::createCamcorderProfile(int cameraId, const char **atts, Vector<in
     const size_t nProfileMappings = sizeof(sCamcorderQualityNameMap)/
             sizeof(sCamcorderQualityNameMap[0]);
     const int quality = findTagForName(sCamcorderQualityNameMap, nProfileMappings, atts[1]);
-    if(quality == -1) return nullptr;
     CHECK(quality != -1);
 
     const size_t nFormatMappings = sizeof(sFileFormatMap)/sizeof(sFileFormatMap[0]);
     const int fileFormat = findTagForName(sFileFormatMap, nFormatMappings, atts[3]);
-    if(fileFormat == -1) return nullptr;
     CHECK(fileFormat != -1);
 
     MediaProfiles::CamcorderProfile *profile = new MediaProfiles::CamcorderProfile;
@@ -437,10 +435,8 @@ MediaProfiles::startElementHandler(void *userData, const char *name, const char
         profiles->mCurrentCameraId = getCameraId(atts);
         profiles->addStartTimeOffset(profiles->mCurrentCameraId, atts);
     } else if (strcmp("EncoderProfile", name) == 0) {
-        MediaProfiles::CamcorderProfile* profile = createCamcorderProfile(profiles->mCurrentCameraId, atts, profiles->mCameraIds);
-        if(profile != nullptr) {
-            profiles->mCamcorderProfiles.add(profile);
-        }
+        profiles->mCamcorderProfiles.add(
+            createCamcorderProfile(profiles->mCurrentCameraId, atts, profiles->mCameraIds));
     } else if (strcmp("ImageEncoding", name) == 0) {
         profiles->addImageEncodingQualityLevel(profiles->mCurrentCameraId, atts);
     }
-- 
2.17.1

